### ISSUE TITLE
Resolve generic classes

### ISSUE THREAD

**Issue by jfantunes** [zijianhuang/webapiclientgen#50]
Hi!
I have several classes that are generic. One of this is the following one:
```
 public class MimsResult<T>
    {
        public T Result { get; set; }
        public DateTime GeneratedAt { get; set; } = DateTime.Now;
        public bool Success { get; set; } = true;
        public string Message { get; set; }
    }

```
For what im observing webapicodegen doesnt like it. When i pass the cherrypick option 1 or 8, he throws me an error  **'System.ArgumentNullException: 'Value cannot be null.'**
Since has you stated before this is simillar to a template it's not a typed class, what can i make it to be more friendly to the codegen witout having major changes on code?
Thanks.

**Comment by zijianhuang** [comment]
@jfantunes , basically done. You may check out this repository and build then test with your .NET Core Web API project.

I had tested with ASP.NET Web API, .net client, jQuery and Angular2+.

If you find things are OK, I will be making a release.

**Comment by jfantunes** [comment]
Hi @zijianhuang, i've tested and works great. Thank you for your time. I'm sure many people would appreciate it, since sometimes generics are the best way to have less Dto's without the same information.

**Comment by zijianhuang** [comment]
released webapiclientgen 2.5.0 and webapiclientgen 2.5.1. tested in https://github.com/zijianhuang/DemoCoreWeb
and https://github.com/zijianhuang/webapiclientgenexamples
